pkg/config/config.go

```diff
@@ -20,14 +20,15 @@ type Serialized struct {
 	Impersonate string `json:"impersonate"`
 }
 
-func (s *Serialized) ToController() (*Controller, error) {
-	cfg := Controller{
-		Report:         s.Report,
-		StoragePath:    s.StoragePath,
-		Endpoint:       s.Endpoint,
-		ReportEndpoint: s.PullReport.Endpoint,
-		Impersonate:    s.Impersonate,
+func (s *Serialized) ToController(cfg *Controller) (*Controller, error) {
```

Comment:
Do we need to send the `cfg` attribute? We can always create a new `Controller` as before or not? Otherwise it looks good.

Reply:
This is called by Load - so the steps are:
Set default to support.Controller (in code)
Try to load config to support.Controller

And unfortunatelly previously loading was resetting whole support.Controller with what was set in config file, so all the defaults had to be sed there. That was ignoring what was already set in support.Controller and creating a new one.

Now, there is an option to not to pass existing Controller and create failback to previous behaviour, but we use the way which passes in the existing support.Controller default values set before.
And there is an option to override the defaults from config. For this there are if len(...)>0 blocks below - if it is set in config file, use it.

Reply:
Ahh I see. Sorry. I should have read the description.